.github/actions/setup-sentry/action.yml

```diff
@@ -0,0 +1,148 @@
+    - name: Setup default environment variables
+      shell: bash
+      env:
+        PYTHON_VERSION: ${{ inputs.python }}
+        NEED_KAFKA: ${{ inputs.kafka }}
+        MATRIX_INSTANCE: ${{ matrix.instance }}
+        MATRIX_INSTANCE_TOTAL: ${{ strategy.job-total }}
+      run: |
+        echo "PIP_DISABLE_PIP_VERSION_CHECK=on" >> $GITHUB_ENV
+        echo "MIGRATIONS_TEST_MIGRATE=0" >> $GITHUB_ENV
+        echo "SENTRY_LIGHT_BUILD=1" >> $GITHUB_ENV
+        echo "SENTRY_SKIP_BACKEND_VALIDATION=1" >> $GITHUB_ENV
+
+        [ "$PYTHON_VERSION" = "3" ] && echo "SENTRY_PYTHON3=1" >> $GITHUB_ENV || true
+
+        ### node configuration ###
+        echo "NODE_ENV=development" >> $GITHUB_ENV
+        echo "NODE_OPTIONS=--max-old-space-size=4096" >> $GITHUB_ENV
+
+        ### pytest-sentry configuration ###
+        echo "PYTEST_SENTRY_DSN=https://6fd5cfea2d4d46b182ad214ac7810508@sentry.io/2423079" >> $GITHUB_ENV
+        echo "PYTEST_ADDOPTS=--reruns 5" >> $GITHUB_ENV
+        # this handles pytest test sharding
+        if [ "$MATRIX_INSTANCE" ]; then
+          echo "TEST_GROUP=$MATRIX_INSTANCE" >> $GITHUB_ENV
+          echo "TOTAL_TEST_GROUPS=$MATRIX_INSTANCE_TOTAL" >> $GITHUB_ENV
+        fi
+        # This records failures on master to sentry in order to detect flakey tests, as it's
+        # expected that people have failing tests on their PRs
+        [ "$GITHUB_REF" = "refs/heads/master" ] && echo "PYTEST_SENTRY_ALWAYS_REPORT=1" >> $GITHUB_ENV || true
+
+        ### services configuration ###
+        # Note: some backend tests (e.g. tests/sentry/eventstream/kafka/test_consumer.py) will behave
+        # differently if these are set.
+        if [ "$NEED_KAFKA" = "true" ]; then
+          echo "SENTRY_KAFKA_HOSTS=127.0.0.1:9092" >> $GITHUB_ENV
+          echo "SENTRY_ZOOKEEPER_HOSTS=127.0.0.1:2181" >> $GITHUB_ENV
+        fi
+
+    # Setup custom pytest matcher, see https://github.com/actions/setup-node/issues/97
+    - name: Add pytest log matcher
+      shell: bash
+      run: |
+        echo "::remove-matcher owner=pytest::"
+        echo "::add-matcher::.github/pytest.json"
+
+    - name: Install system dependencies
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y --no-install-recommends \
+          libxmlsec1-dev \
+          libmaxminddb-dev
+
+    - name: Set up outputs
+      id: config
+      env:
+        MATRIX_INSTANCE: ${{ matrix.instance }}
+      shell: bash
+      run: |
+        echo "::set-output name=yarn-cache-dir::$(yarn cache dir)"
+        echo "::set-output name=matrix-instance-number::$(($MATRIX_INSTANCE+1))"
+        echo "::set-output name=acceptance-dir::.artifacts/visual-snapshots/acceptance"
+
+    - name: Install pip
+      shell: bash
+      run: |
+        pip install --no-cache-dir --upgrade "pip>=20.0.2"
+
+    - name: Get pip cache dir
+      id: pip-cache
+      shell: bash
+      run: |
+        echo "::set-output name=pip-cache::$(pip cache dir)"
+
```

Comment:
@billyvg Can you follow up by restoring cache here? I definitely assumed the cache action was to upload, not download, for some reason. And didn't bother to read job output on the pip install step. (It's always Downloading now.)

related conversation for posterity https://github.com/getsentry/sentry/pull/21286#discussion_r503606908

Reply:
https://github.com/getsentry/sentry/pull/21295